internal/k8s/reconciler/route.go

```diff
@@ -360,10 +361,20 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 	switch route := r.Route.(type) {
 	case *gw.HTTPRoute:
 		for _, httpRule := range route.Spec.Rules {
+			r.logger.Warn("starting validation...")
 			rule := httpRule
 			routeRule := service.NewRouteRule(&rule)
+
 			for _, backendRef := range rule.BackendRefs {
 				ref := backendRef
+
+				allowed, err := routeAllowedForBackendRef(ctx, r.Route, ref.BackendRef, r.client)
+				if err != nil || !allowed {
+					// FUTURE Set appropriate status
+					r.logger.Warn("Cross-namespace routing not allowed, will enforce in future", "refName", ref.Name, "refNamespace", ref.Namespace)
+					continue
```

Comment:
Going back and forth myself over letting this fall through to the resolver below for now so we don't have any functional impact with the merge of this PR vs. early out and just not setting the status until followup PR. Thoughts @andrewstucki @mikemorris @sarahalsmiller ?

Reply:
Yeah, I'd just let this fall through for now if we're not going to set the status until the follow-up so that the functionality actually gets changed in a single commit -- the `continue` here  _should_ cause some silent routing failures for a bit since the route will have no errors on it but also no resolved services that it can write into Consul once it binds to a gateway.

Also, before I do a full-blown formal review, just to mention, you should probably return if `err != nil` since the only thing that could cause that in the underlying code is a Kubernetes client error (i.e. a network issue, api server issue, auth issue, etc., all of which we _do_ want to reschedule for reconciliation) -- it's only if we can actually determine that the route isn't allowed to reference tthe given `backendRefs` values that we should mark the Status as `RefNotAllowed`.

Reply:
Done and done